x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/cluster/routing/allocation/mapper/DataTierFieldMapper.java

```diff
@@ -50,6 +50,12 @@ public String familyTypeName() {
             return KeywordFieldMapper.CONTENT_TYPE;
         }
 
+        @Override
+        public boolean isAggregatable() {
+            // Doesn't support field data (yet) and therefor attempting to aggregate on field of this field type fail
```

Comment:
The field caps api loads index field data to determine a field's value source type.
This field type doesn't have field data support (it doesn't overwrite `fielddataBuilder(...)` method).
So this field of this type never really supported aggregations, and there for returning `false` here is ok.